datafusion/src/physical_plan/mod.rs

```diff
@@ -570,6 +570,7 @@ pub trait Accumulator: Send + Sync + Debug {
 pub mod aggregates;
 pub mod analyze;
 pub mod array_expressions;
+pub mod avro;
```

Comment:
This should also be feature-flagged so that the flags inside its file can be removed. There's no use compiling the avro module if the flag is disabled.

Reply:
That implies adding the avro feature flag in a number of other places such as execution/context.rs, is that what you are asking for implicitly ?

Reply:
I think @nevi-me  is suggesting something like

```suggestion
#[cfg(feature = "avro")]
pub mod avro;
```

Which would then let you avoid so much `#[cfg...]` in https://github.com/apache/arrow-datafusion/pull/910#discussion_r700593118

I am not sure what other changes that entails

Reply:
@alamb The physical plan is used everywhere in the codebase, so feature gating in the avro submod of physical plan seemed to be the best thing to do